dev/Bootstrap/CS/Microsoft.WindowsAppSDK.Bootstrap.Net/Bootstrap.cs

```diff
@@ -0,0 +1,218 @@
+        /// @see Initialize(uint, string)
+        /// @see Initialize(uint, string, PackageVersion)
+        /// @see Shutdown()
+        public static void Initialize(uint majorMinorVersion)
+        {
+            Initialize(majorMinorVersion, null);
+        }
+
+        /// Initialize the calling process to use Windows App SDK's framework package.
+        ///
+        /// Find a Windows App SDK framework package meeting the criteria and make it available
+        /// for use by the current process. If multiple packages meet the criteria the best
+        /// candidate is selected.
+        ///
+        /// This is equivalent to `Initialize(majorMinorVersion, versionTag, new PackageVersion())`.
+        ///
+        /// @param majorMinorVersion major and minor version of Windows App SDK's framework package, encoded as `0xMMMMNNNN` where M=Major, N=Minor (e.g. 1.2 == 0x00010002).
+        /// @param versionTag version tag (if any), e.g. "preview1".
+        /// @see Initialize(uint)
+        /// @see Initialize(uint, string, PackageVersion)
+        /// @see Shutdown()
+        public static void Initialize(uint majorMinorVersion, string versionTag)
+        {
+            Initialize(majorMinorVersion, versionTag, new PackageVersion());
+        }
+
+        /// Initialize the calling process to use Windows App SDK's framework package.
+        ///
+        /// Find a Windows App SDK framework package meeting the criteria and make it available
+        /// for use by the current process. If multiple packages meet the criteria the best
+        /// candidate is selected.
+        ///
+        /// @param majorMinorVersion major and minor version of Windows App SDK's framework package, encoded as `0xMMMMNNNN` where M=Major, N=Minor (e.g. 1.2 == 0x00010002).
+        /// @param versionTag version tag (if any), e.g. "preview1".
+        /// @param minVersion the minimum version to use
+        /// @see Initialize(uint)
+        /// @see Initialize(uint, string)
+        /// @see Shutdown()
+        public static void Initialize(uint majorMinorVersion, string versionTag, PackageVersion minVersion)
+        {
+            NativeMethods.MddBootstrapInitialize_Throw(majorMinorVersion, versionTag, minVersion);
+        }
+
+        /// Undo the changes made by Initialize().
+        ///
+        /// @warning Packages made available via `Initialize()` and
+        ///          the Dynamic Dependencies API should not be used after this call.
+        /// @see Initialize(uint)
+        /// @see Initialize(uint, string)
+        /// @see Initialize(uint, string, PackageVersion)
+        public static void Shutdown()
+        {
+            NativeMethods.MddBootstrapShutdown();
+        }
+    }
+
+    // The Windows App SDK bootstrap initialization API. This class returns error as an HRESULT
+    // (>=0 on success, <0 on error).
+    //
+    // @see Bootstrap
+    public class Bootstrap_NoThrow
```

Comment:
Basically everything in C#-land should use exceptions, not hresults. What's the motivation for having this class? I'd just delete it. If you wanted to have a version that returned "nuh-uh, can't init" then have a method like TryInitialize instead, and it returns a `bool` instead of an `int`.

Reply:
`PreserveSig` is `true` by default. Folks calling into an wrapper of unmanaged Win32 `noexcept` APIs may be surprised by a sudden exception created out of thin air.

I'm okay with the deletion if you provide the `TryInitialize` variant.

Reply:
Idiomatic = +1

TryInitialize = Good idea, but return boolean means you lose the HRESULT. Which would you suggest:

#1

```c#
public static bool TryInitialize(..., out int hresult)
```

#2

```c#
public static int TryInitialize(...)
```

#3

```c#
public static int Initialize_NoThrow(...)
```

or ?

Reply:
I like `#1`!

Reply:
Nope, int is correct. HRESULT is a typedef for LONG i.e. 32-bit signed integer. Hence the reason you often seen values as ginormous negative integers

Hex at least is less cryptic. `0x80070005` == `-2147024891` == `E_ACCESS_DENIED`. Or maybe I've been doing this too long I don't see integers I just see severity, facility, code ;-)

Reply:
I like #1 too. Very C#esque e.g. Int.TryParse.

Thanks @jonwis and @riverar for the feedback!

See latest commit for revised syntax